File: Data Structures/Graph-based-DSAs/Adjacency-matrix.cpp
/*Adjacency matrix implementation in cpp:*/

#include "iostream"

using namespace std;

class Graph{
    private:
        bool ** adjMatrix = nullptr;
        int numVertices;
    public:
        Graph(int numVertices){
            this->numVertices = numVertices;
            adjMatrix = new bool * [numVertices];
            for (int i = 0; i < numVertices; i++){
                adjMatrix[i] = new bool [numVertices];
                for (int j= 0; j < numVertices; j++){
                    adjMatrix[i][j] = false;
                }
            }

        }

        void add_edge(int i, int j){
            adjMatrix[i][j] = true;
            adjMatrix[j][i] = true;
        }

        void remove_edge(int i, int j){
            adjMatrix[i][j] = false;
            adjMatrix[j][i] = false;
        }

        void print_matrix(){
            for (int i = 0; i < numVertices; i ++){
                cout << i << " : ";
                for (int j = 0; j < numVertices; j++){
                    cout << adjMatrix[i][j] << " ";
                }
                cout << endl;
            }
        }

        ~Graph(){
            for (int i = 0; i < numVertices ; i++){
                delete[] adjMatrix[i];
            }
            delete[] adjMatrix; 
        }
};

int main(){
    Graph g(5);
    g.add_edge(0, 1);
    g.add_edge(0, 2);
    g.add_edge(1, 2);
    g.add_edge(2, 0);
    g.add_edge(2, 3);
    g.print_matrix();
}

File: Data Structures/Graph-based-DSAs/Adjacency-matrix.py
"""Adjacency matrix representation in python:"""

class Graph:
    #initialize the matrix:
    def __init__(self, size):
        self.adjMatrix = []
        self.size = size
        for i in range(size):
            self.adjMatrix.append([0 for i in range(size)])

    def add_edge(self, v1, v2):
        if v1 == v2:
            print("Same vertex %d and %d" % (v1, v2))
            return
        self.adjMatrix[v1][v2] = True
        self.adjMatrix[v2][v1] = True
    
    def remove_edges(self, v1, v2):
        if not self.adjMatrix[v1, v2]:
            print("No edge between %d and %d" % (v1, v2))
            return
        self.adjMatrix[v1, v2] = False
        self.adjMatrix[v2, v1] = False
    def __len__(self):
        return self.size
    def print_matrix(self):
        for row in self.adjMatrix:
            for val in row:
                print('{: 4}'.format(val), end=''),
            print("")

    
if __name__ == "__main__":
    g = Graph(5)
    g.add_edge(0, 1)
    g.add_edge(0, 2)
    g.add_edge(1, 2)
    g.add_edge(2, 0)
    g.add_edge(2, 3)

    g.print_matrix()